### PR TITLE
Added the backend for handeling isApproved

### PR DESCRIPTION
**Pull request by eanker**
Backend for preview-code/minimal-frontend#43

TODO: Setting the status on github, but that will be a separate pull request.

### CODE REVIEW

File: src/main/java/previewcode/backend/DTO/ApprovedGroup.java [hunk 1]
```diff
@@ -0,0 +1,17 @@
...
+
+    /**
+     * If the group is approved
+     */
+    public String approved;
+
+    /**
+     * All the hunks in this group
+     */
+    public Map<String, String> hunks;
```
Comment (txsmith) [inline src/main/java/previewcode/backend/DTO/ApprovedGroup.java]: Can you add `@JsonProperty` annotations here?

**Review by txsmith — CHANGES_REQUESTED** [review]

File: src/main/java/previewcode/backend/DTO/ApprovedGroup.java (hunk 1 above)
Comment (txsmith) [reply src/main/java/previewcode/backend/DTO/ApprovedGroup.java]: And a constructor with `@JsonCreator` and `@JsonProperty` on the arguments

**Review by txsmith — CHANGES_REQUESTED** [review]

File: src/main/java/previewcode/backend/DTO/ApprovedGroup.java [hunk 2]
```diff
@@ -0,0 +1,17 @@
+package previewcode.backend.DTO;
+
+import java.util.Map;
+
+public class ApprovedGroup {
+
+    /**
+     * If the group is approved
+     */
+    public String approved;
```
Comment (txsmith) [inline src/main/java/previewcode/backend/DTO/ApprovedGroup.java]: Should be `Option<Boolean>`

File: src/main/java/previewcode/backend/DTO/IsApproved.java [hunk 3]
```diff
@@ -0,0 +1,20 @@
...
+import java.util.Map;
+
+/**
+ * Information if a pull request or it's groups/hunks are approved
+ */
+public class IsApproved {
+    /**
+     * If the pull request is approved
+     */
+    public String approved;
```
Comment (txsmith) [inline src/main/java/previewcode/backend/DTO/IsApproved.java]: Should be `Option<Boolean>`

**Review by txsmith — CHANGES_REQUESTED** [review]

**Comment by eanker** [comment]
replaced by #52

**Pull request closed without merging**